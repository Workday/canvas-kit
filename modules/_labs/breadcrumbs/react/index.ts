import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
