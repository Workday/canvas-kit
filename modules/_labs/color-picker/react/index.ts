export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
