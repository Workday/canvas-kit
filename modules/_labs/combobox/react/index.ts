import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
