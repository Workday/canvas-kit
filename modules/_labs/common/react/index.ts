export * from './lib/Flex';
export * from './lib/Box';
