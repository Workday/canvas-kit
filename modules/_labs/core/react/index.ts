import type from './lib/type';
import space from './lib/space';
export default type;
export {type, space};
export * from './lib/type';
export * from './lib/StaticStates';
