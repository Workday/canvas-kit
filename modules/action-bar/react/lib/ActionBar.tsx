import * as React from 'react';
import styled from 'react-emotion';
import {colors, commonColors, spacing} from '@workday/canvas-kit-react-core';

export interface ActionBarProps extends React.HTMLAttributes<HTMLDivElement> {
  /**
   * If the actionBar is fixed to the bottom of the screen
   */
  fixed?: boolean;
}

const ActionBarContainer = styled('div')(
  {
    borderTop: `solid 1px ${colors.soap400}`,
    background: commonColors.background,
    padding: spacing.s,
    boxShadow: '0 -2px 4px rgba(0, 0, 0, 0.08)',
    '@media (max-width: 575px)': {
      padding: spacing.xxs,
    },
  },
  ({fixed}: ActionBarProps) =>
    fixed && {
      position: 'fixed',
      left: 0,
      bottom: 0,
      right: 0,
    }
);

const ChildrenContainer = styled('div')({
  display: 'inline-block',
  padding: `0 ${spacing.m}`,
  '*:not(:first-child)': {
    marginLeft: spacing.s,
  },
  '@media (max-width: 575px)': {
    display: 'flex',
    padding: spacing.xxs,
    justifyContent: 'center',
    flexDirection: 'row-reverse',
    '> *': {
      flex: 1,
      '&:not(:first-child)': {
        marginRight: spacing.s,
        marginLeft: 0,
      },
    },
  },
});

export default class ActionBar extends React.Component<ActionBarProps> {
  public render() {
    const {fixed, children, ...props} = this.props;

    return (
      <ActionBarContainer {...props} fixed={fixed}>
        <ChildrenContainer>{children}</ChildrenContainer>
      </ActionBarContainer>
    );
  }
}
