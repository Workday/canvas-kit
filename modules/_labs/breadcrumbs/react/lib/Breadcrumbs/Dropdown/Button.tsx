/** @jsx jsx */
import React from 'react';
import {css, jsx} from '@emotion/core';
import {IconButton, IconButtonProps} from '@workday/canvas-kit-react-button';
import {folderCloseIcon} from '@workday/canvas-system-icons-web';
import {colors} from '@workday/canvas-kit-react-core';

interface DropdownButtonProps extends IconButtonProps {
  buttonRef: React.Ref<HTMLButtonElement>;
  setIsDropdownOpen: React.Dispatch<React.SetStateAction<boolean>>;
  toggled: boolean;
}

export const DropdownButton = ({
  buttonRef,
  setIsDropdownOpen,
  toggled,
  variant = IconButton.Variant.Plain,
  'aria-label': ariaLabel = 'More links',
  ...props
}: DropdownButtonProps) => {
  const hasPlainVariant = variant === IconButton.Variant.Plain;
  // gives the plain icon variant a little extra space between the chevron separators
  const customStyles = css(hasPlainVariant ? {margin: '0 1px'} : {});

  return (
    <IconButton
      css={customStyles}
      variant={variant}
      icon={folderCloseIcon}
      color={colors.licorice200}
      buttonRef={buttonRef}
      toggled={toggled}
      onClick={() => setIsDropdownOpen(!toggled)}
      aria-label={ariaLabel}
      aria-pressed={undefined}
      aria-expanded={toggled}
      aria-haspopup={true}
      aria-controls="menu"
      {...props}
    />
  );
};
