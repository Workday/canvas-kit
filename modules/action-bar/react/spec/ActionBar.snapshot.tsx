import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Button from '@workday/canvas-kit-react-button';
import ActionBar from '../index';

describe('Action Bar Snapshots', () => {
  test('renders an empty action bar', () => {
    const component = renderer.create(<ActionBar />);
    expect(component).toMatchSnapshot();
  });

  test('renders a fixed empty action bar', () => {
    const component = renderer.create(<ActionBar fixed={true} />);
    expect(component).toMatchSnapshot();
  });

  test('renders an action bar with one button', () => {
    const component = renderer.create(
      <ActionBar>
        <Button variant={Button.Variant.Primary}>Button</Button>
      </ActionBar>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders a fixed action bar with one button', () => {
    const component = renderer.create(
      <ActionBar fixed={true}>
        <Button variant={Button.Variant.Primary}>Button</Button>
      </ActionBar>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders an action bar with two buttons', () => {
    const component = renderer.create(
      <ActionBar>
        <Button variant={Button.Variant.Primary}>Button</Button>
        <Button>Button</Button>
      </ActionBar>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders a fixed action bar with two buttons', () => {
    const component = renderer.create(
      <ActionBar fixed={true}>
        <Button variant={Button.Variant.Primary}>Button</Button>
        <Button>Button</Button>
      </ActionBar>
    );
    expect(component).toMatchSnapshot();
  });
});
