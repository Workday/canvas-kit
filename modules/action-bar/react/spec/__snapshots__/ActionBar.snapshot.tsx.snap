// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Bar Snapshots renders a fixed action bar with one button 1`] = `
.emotion-4 {
  display: inline-block;
  padding: 0 24px;
}

.emotion-4 *:not(:first-child) {
  margin-left: 16px;
}

@media (max-width:575px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding: 8px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:first-child) {
    margin-right: 16px;
    margin-left: 0;
  }
}

.emotion-6 {
  border-top: solid 1px #dfe2e6;
  background: #ffffff;
  padding: 16px;
  box-shadow: 0 -2px 4px rgba(0,0,0,0.08);
  position: fixed;
  left: 0;
  bottom: 0;
  right: 0;
}

@media (max-width:575px) {
  .emotion-6 {
    padding: 8px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
  height: 40px;
  padding: 0 32px;
  min-width: 112px;
  max-width: 288px;
  font-weight: 500;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2:focus {
  background-color: #ffa126;
  border-color: #0875e1;
  color: #ffffff;
}

.emotion-2:hover:focus {
  background-color: #f38b00;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:hover {
  background-color: #f38b00;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #fcd49f;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:not([disabled]):focus,
.emotion-2:not([disabled]):active {
  border-color: #0875e1;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  padding: 0 12px;
  font-size: inherit;
  font-weight: inherit;
  padding: 0px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <button
      className="emotion-2 emotion-3"
    >
      <span
        className="emotion-0 emotion-1"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Action Bar Snapshots renders a fixed action bar with two buttons 1`] = `
.emotion-8 {
  display: inline-block;
  padding: 0 24px;
}

.emotion-8 *:not(:first-child) {
  margin-left: 16px;
}

@media (max-width:575px) {
  .emotion-8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding: 8px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .emotion-8 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-8 > *:not(:first-child) {
    margin-right: 16px;
    margin-left: 0;
  }
}

.emotion-10 {
  border-top: solid 1px #dfe2e6;
  background: #ffffff;
  padding: 16px;
  box-shadow: 0 -2px 4px rgba(0,0,0,0.08);
  position: fixed;
  left: 0;
  bottom: 0;
  right: 0;
}

@media (max-width:575px) {
  .emotion-10 {
    padding: 8px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
  height: 40px;
  padding: 0 32px;
  min-width: 112px;
  max-width: 288px;
  font-weight: 500;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2:focus {
  background-color: #ffa126;
  border-color: #0875e1;
  color: #ffffff;
}

.emotion-2:hover:focus {
  background-color: #f38b00;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:hover {
  background-color: #f38b00;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #fcd49f;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:not([disabled]):focus,
.emotion-2:not([disabled]):active {
  border-color: #0875e1;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  padding: 0 12px;
  font-size: inherit;
  font-weight: inherit;
  padding: 0px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  background-color: #f0f1f2;
  border-color: #ced3d9;
  color: #333333;
  height: 40px;
  padding: 0 32px;
  min-width: 112px;
  max-width: 288px;
  font-weight: 500;
}

.emotion-6:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-6:disabled,
.emotion-6:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-6:focus {
  background-color: #f0f1f2;
  border-color: #0875e1;
  color: #333333;
}

.emotion-6:hover:focus {
  background-color: #ced3d9;
}

.emotion-6:active,
.emotion-6:focus:active,
.emotion-6:hover:active {
  background-color: #f0f1f2;
  border-color: #ced3d9;
  color: #333333;
}

.emotion-6:hover {
  background-color: #ced3d9;
  border-color: #a1aab3;
  color: #333333;
}

.emotion-6:disabled,
.emotion-6:active:disabled,
.emotion-6:focus:disabled,
.emotion-6:hover:disabled {
  background-color: #ffffff;
  border-color: #ced3d9;
  color: #ced3d9;
}

.emotion-6:not([disabled]):focus,
.emotion-6:not([disabled]):active {
  border-color: #0875e1;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

<div
  className="emotion-10 emotion-11"
>
  <div
    className="emotion-8 emotion-9"
  >
    <button
      className="emotion-2 emotion-3"
    >
      <span
        className="emotion-0 emotion-1"
      >
        Button
      </span>
    </button>
    <button
      className="emotion-6 emotion-3"
    >
      <span
        className="emotion-0 emotion-1"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Action Bar Snapshots renders a fixed empty action bar 1`] = `
.emotion-0 {
  display: inline-block;
  padding: 0 24px;
}

.emotion-0 *:not(:first-child) {
  margin-left: 16px;
}

@media (max-width:575px) {
  .emotion-0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding: 8px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .emotion-0 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0 > *:not(:first-child) {
    margin-right: 16px;
    margin-left: 0;
  }
}

.emotion-2 {
  border-top: solid 1px #dfe2e6;
  background: #ffffff;
  padding: 16px;
  box-shadow: 0 -2px 4px rgba(0,0,0,0.08);
  position: fixed;
  left: 0;
  bottom: 0;
  right: 0;
}

@media (max-width:575px) {
  .emotion-2 {
    padding: 8px;
  }
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  />
</div>
`;

exports[`Action Bar Snapshots renders an action bar with one button 1`] = `
.emotion-6 {
  border-top: solid 1px #dfe2e6;
  background: #ffffff;
  padding: 16px;
  box-shadow: 0 -2px 4px rgba(0,0,0,0.08);
}

@media (max-width:575px) {
  .emotion-6 {
    padding: 8px;
  }
}

.emotion-4 {
  display: inline-block;
  padding: 0 24px;
}

.emotion-4 *:not(:first-child) {
  margin-left: 16px;
}

@media (max-width:575px) {
  .emotion-4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding: 8px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .emotion-4 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-4 > *:not(:first-child) {
    margin-right: 16px;
    margin-left: 0;
  }
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
  height: 40px;
  padding: 0 32px;
  min-width: 112px;
  max-width: 288px;
  font-weight: 500;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2:focus {
  background-color: #ffa126;
  border-color: #0875e1;
  color: #ffffff;
}

.emotion-2:hover:focus {
  background-color: #f38b00;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:hover {
  background-color: #f38b00;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #fcd49f;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:not([disabled]):focus,
.emotion-2:not([disabled]):active {
  border-color: #0875e1;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  padding: 0 12px;
  font-size: inherit;
  font-weight: inherit;
  padding: 0px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-4 emotion-5"
  >
    <button
      className="emotion-2 emotion-3"
    >
      <span
        className="emotion-0 emotion-1"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Action Bar Snapshots renders an action bar with two buttons 1`] = `
.emotion-10 {
  border-top: solid 1px #dfe2e6;
  background: #ffffff;
  padding: 16px;
  box-shadow: 0 -2px 4px rgba(0,0,0,0.08);
}

@media (max-width:575px) {
  .emotion-10 {
    padding: 8px;
  }
}

.emotion-8 {
  display: inline-block;
  padding: 0 24px;
}

.emotion-8 *:not(:first-child) {
  margin-left: 16px;
}

@media (max-width:575px) {
  .emotion-8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding: 8px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .emotion-8 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-8 > *:not(:first-child) {
    margin-right: 16px;
    margin-left: 0;
  }
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
  height: 40px;
  padding: 0 32px;
  min-width: 112px;
  max-width: 288px;
  font-weight: 500;
}

.emotion-2:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-2:disabled,
.emotion-2:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-2:focus {
  background-color: #ffa126;
  border-color: #0875e1;
  color: #ffffff;
}

.emotion-2:hover:focus {
  background-color: #f38b00;
}

.emotion-2:active,
.emotion-2:focus:active,
.emotion-2:hover:active {
  background-color: #ffa126;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:hover {
  background-color: #f38b00;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:disabled,
.emotion-2:active:disabled,
.emotion-2:focus:disabled,
.emotion-2:hover:disabled {
  background-color: #fcd49f;
  border-color: #ffa126;
  color: #ffffff;
}

.emotion-2:not([disabled]):focus,
.emotion-2:not([disabled]):active {
  border-color: #0875e1;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  padding: 0 12px;
  font-size: inherit;
  font-weight: inherit;
  padding: 0px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  background-color: #f0f1f2;
  border-color: #ced3d9;
  color: #333333;
  height: 40px;
  padding: 0 32px;
  min-width: 112px;
  max-width: 288px;
  font-weight: 500;
}

.emotion-6:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-6:disabled,
.emotion-6:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-6:focus {
  background-color: #f0f1f2;
  border-color: #0875e1;
  color: #333333;
}

.emotion-6:hover:focus {
  background-color: #ced3d9;
}

.emotion-6:active,
.emotion-6:focus:active,
.emotion-6:hover:active {
  background-color: #f0f1f2;
  border-color: #ced3d9;
  color: #333333;
}

.emotion-6:hover {
  background-color: #ced3d9;
  border-color: #a1aab3;
  color: #333333;
}

.emotion-6:disabled,
.emotion-6:active:disabled,
.emotion-6:focus:disabled,
.emotion-6:hover:disabled {
  background-color: #ffffff;
  border-color: #ced3d9;
  color: #ced3d9;
}

.emotion-6:not([disabled]):focus,
.emotion-6:not([disabled]):active {
  border-color: #0875e1;
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

<div
  className="emotion-10 emotion-11"
>
  <div
    className="emotion-8 emotion-9"
  >
    <button
      className="emotion-2 emotion-3"
    >
      <span
        className="emotion-0 emotion-1"
      >
        Button
      </span>
    </button>
    <button
      className="emotion-6 emotion-3"
    >
      <span
        className="emotion-0 emotion-1"
      >
        Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Action Bar Snapshots renders an empty action bar 1`] = `
.emotion-2 {
  border-top: solid 1px #dfe2e6;
  background: #ffffff;
  padding: 16px;
  box-shadow: 0 -2px 4px rgba(0,0,0,0.08);
}

@media (max-width:575px) {
  .emotion-2 {
    padding: 8px;
  }
}

.emotion-0 {
  display: inline-block;
  padding: 0 24px;
}

.emotion-0 *:not(:first-child) {
  margin-left: 16px;
}

@media (max-width:575px) {
  .emotion-0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    padding: 8px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .emotion-0 > * {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .emotion-0 > *:not(:first-child) {
    margin-right: 16px;
    margin-left: 0;
  }
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  />
</div>
`;
